MvvmCross/Core/Core/ViewModels/MvxViewModel.cs

```diff
@@ -99,11 +100,16 @@ public abstract class MvxViewModel<TParameter> : MvxViewModel, IMvxViewModel<TPa
     public abstract class MvxViewModelResult<TResult> : MvxViewModel, IMvxViewModelResult<TResult> where TResult : class
     {
         TaskCompletionSource<TResult> _tcs;
+        CancellationToken _cancellationToken;
         bool _isClosing;
 
-        public void SetClose(TaskCompletionSource<TResult> tcs)
+        public void SetClose(TaskCompletionSource<TResult> tcs, CancellationToken cancellationToken)
         {
             _tcs = tcs ?? throw new ArgumentNullException(nameof(tcs));
```

Comment:
I thought of that as well, but the Dispose method will take care of that already.